### PR TITLE
v1.13 update

### LINKED ISSUE #412: Update deployed k8s version to v1.13
/kind feature
/priority important-soon

**Describe the solution you'd like**

Currently we are deploying v1.12 clusters, for kubeadm HA support we will need a min. version of v1.13.

Fix for this will require the following:

- Update the image building tooling for any v1.13 related changes
- Build new images for v1.13 and update the documentation for the newly built images
- Update kubeadm configs to use the v1beta1 structs rather than v1alpha3 structs
- Update examples to reference 1.13 instead of 1.12

### PR DESCRIPTION
**Pull request by chuckha**
* Updates AMIs and related functions
* Updates kubeadm to use v1beta1 types

Signed-off-by: Chuck Ha <chuck@heptio.com>

Fixes #412 

```release-note
Updates kubernetes cluster version to v1.13.0
```

### CODE REVIEW

File: build/amis/README.md [hunk 1]
```diff
@@ -125,5 +125,5 @@ Connect remotely to an instance created from the image and run the Node Conforma
 wget https://dl.k8s.io/$(< /etc/kubernetes_community_ami_version)/kubernetes-test.tar.gz
 tar -zxvf kubernetes-test.tar.gz kubernetes/platforms/linux/amd64
 cd kubernetes/platforms/linux/amd64
-sudo ./ginkgo --nodes=8 --flakeAttempts=2 --focus="\[Conformance\]" --skip="\[Flaky\]|\[Serial\]|\[sig-network\]|Container Lifecycle Hook" ./e2e_node.test -- --k8s-bin-dir=/usr/bin
-```
\ No newline at end of file
+sudo ./ginkgo --nodes=1 --flakeAttempts=2 --focus="\[Conformance\]" --skip="\[Flaky\]|\[Serial\]|\[sig-network\]|Container Lifecycle Hook" ./e2e_node.test -- --k8s-bin-dir=/usr/bin
```
Comment (vincepri) [inline build/amis/README.md]: Why was this changed?

File: pkg/cloud/aws/services/userdata/controlplane.go [hunk 2]
```diff
@@ -26,13 +26,14 @@ HOSTNAME="$(curl http://169.254.169.254/latest/meta-data/local-hostname)"
...
-  - "$PRIVATE_IP"
-  - "{{.ELBAddress}}"
-apiServerExtraArgs:
-  cloud-provider: aws
+apiServer:
+  certSANs:
+    - "$PRIVATE_IP"
+    - "{{.ELBAddress}}"
+  extraArgs:
+    cloud-provider: aws
```
Comment (vincepri) [inline pkg/cloud/aws/services/userdata/controlplane.go]: The new beta1 config looks 👌

File: build/amis/README.md (hunk 1 above)
Comment (chuckha) [reply build/amis/README.md]: this doesn't have to change. I haven't been able to get to this to work at all. I thought it was running when I sumbitted the PR but it ended up running but not able to schedule any pods. Need to configure *something* but i'm still debugging.

**Comment by chuckha** [comment]
/hold

figured out the problem. updating docs.
But good news!

```
Ran 113 of 284 Specs in 429.215 seconds
SUCCESS! -- 113 Passed | 0 Failed | 0 Flaked | 0 Pending | 171 Skipped 
```

**Comment by chuckha** [comment]
/hold cancel

All issues fixed and instructions copy/pasted to a successful run

Ran 113 of 284 Specs in 540.128 seconds
SUCCESS! -- 113 Passed | 0 Failed | 0 Flaked | 0 Pending | 171 Skipped 

Ginkgo ran 1 suite in 9m1.717261913s
Test Suite Passed

File: build/amis/README.md [hunk 3]
```diff
@@ -121,9 +131,49 @@ packer build -var-file base-images-us-east-1.json -var ami_regions='us-west-2'
...
+As a non-root user:
+
 ```sh
 wget https://dl.k8s.io/$(< /etc/kubernetes_community_ami_version)/kubernetes-test.tar.gz
 tar -zxvf kubernetes-test.tar.gz kubernetes/platforms/linux/amd64
 cd kubernetes/platforms/linux/amd64
-sudo ./ginkgo --nodes=8 --flakeAttempts=2 --focus="\[Conformance\]" --skip="\[Flaky\]|\[Serial\]|\[sig-network\]|Container Lifecycle Hook" ./e2e_node.test -- --k8s-bin-dir=/usr/bin
-```
\ No newline at end of file
+sudo ./ginkgo --nodes=8 --flakeAttempts=2 --focus="\[Conformance\]" --skip="\[Flaky\]|\[Serial\]|\[sig-network\]|Container Lifecycle Hook" ./e2e_node.test -- --k8s-bin-dir=/usr/bin --container-runtime=remote --container-runtime-endpoint unix:///var/run/containerd/containerd.sock --container-runtime-process-name /usr/local/bin/containerd --container-runtime-pid-file= --kubelet-flags="--cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/containerd.service" --extra-log="{\"name\": \"containerd.log\", \"journalctl\": [\"-u\", \"containerd\"]}"
```
Comment (detiber) [inline build/amis/README.md]: Now that a CNI is being initialized, we can probably remove the skip for `\[sig-network\]` and `Container Lifecycle Hook`, since those were previously added to work around the lack of CNI :)

File: build/amis/README.md [hunk 4]
```diff
@@ -121,9 +131,49 @@ packer build -var-file base-images-us-east-1.json -var ami_regions='us-west-2'
 
 Connect remotely to an instance created from the image and run the Node Conformance tests using the following commands:
 
+### Initialize a CNI
```
Comment (ashish-amarnath) [inline build/amis/README.md]: When are these tests run?
Does it make sense to add these steps into a script that can be run? That might be a step forward towards some of the test automation that we have open issues for. WDYT?

Comment (detiber) [reply build/amis/README.md]: Eventually, yes. Currently these tests are only run to validate a new AMI image, which is still a manual step.
